Update BigQuery dependencies to support HOURLY partitioning of tables

wendigo opened the pull request:
I did not update to latest BOM version to minimize the impact of this change.

Hourly partitioning was added in https://github.com/googleapis/java-bigquery/commit/90f998040bbca5882ac3dbcdb4b157f26f48eb01#diff-3dfcf884473c6d47c7c3debc58ce20c0

losipiuk commented:
LGTM - it would be great to add some test coverage for this one. PT?

findepi commented:
@losipiuk We have no tests for BigQuery currently.
cc @davidrabinowitz

losipiuk commented:
> @losipiuk We have no tests for BigQuery currently.

Yeah - I know :)

> cc @davidrabinowitz

ebyhr commented:
Though I'm testing this PR manually, it throws the following exception.
```
 CREATE TABLE
   ebyhr.t4968d (transaction_id INT64,
     transaction_ts TIMESTAMP)
 PARTITION BY
   TIMESTAMP_TRUNC(transaction_ts, HOUR)
```

```
presto> select * from bigquery.ebyhr.t4968d;
Query 20200825_131245_00004_y767b failed: Illegal Argument - Got unexpected time partitioning {"field":"transaction_ts","type":"HOUR"} in project ivory-oarlock-268702 in dataset ebyhr in table t4968d
java.lang.IllegalArgumentException: Illegal Argument - Got unexpected time partitioning {"field":"transaction_ts","type":"HOUR"} in project ivory-oarlock-268702 in dataset ebyhr in table t4968d
	at com.google.cloud.bigquery.StandardTableDefinition.fromPb(StandardTableDefinition.java:286)
	at com.google.cloud.bigquery.TableDefinition.fromPb(TableDefinition.java:159)
	at com.google.cloud.bigquery.TableInfo$BuilderImpl.<init>(TableInfo.java:195)
	at com.google.cloud.bigquery.Table.fromPb(Table.java:630)
	at com.google.cloud.bigquery.BigQueryImpl.getTable(BigQueryImpl.java:682)
	at io.prestosql.plugin.bigquery.BigQueryClient.getTable(BigQueryClient.java:73)
	at io.prestosql.plugin.bigquery.BigQueryMetadata.getBigQueryTable(BigQueryMetadata.java:136)
	at io.prestosql.plugin.bigquery.BigQueryMetadata.getTableHandle(BigQueryMetadata.java:125)
	at io.prestosql.metadata.MetadataManager.getTableHandle(MetadataManager.java:343)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:1055)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:315)
	at io.prestosql.sql.tree.Table.accept(Table.java:53)
	at io.prestosql.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:332)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:2263)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:1306)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:315)
	at io.prestosql.sql.tree.QuerySpecification.accept(QuerySpecification.java:144)
	at io.prestosql.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:332)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:342)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:929)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:315)
	at io.prestosql.sql.tree.Query.accept(Query.java:107)
	at io.prestosql.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.prestosql.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:332)
	at io.prestosql.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:301)
	at io.prestosql.sql.analyzer.Analyzer.analyze(Analyzer.java:83)
	at io.prestosql.sql.analyzer.Analyzer.analyze(Analyzer.java:75)
	at io.prestosql.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:252)
	at io.prestosql.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:181)
	at io.prestosql.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:752)
	at io.prestosql.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:123)
	at io.prestosql.$gen.Presto_testversion____20200825_130926_3.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: No enum constant com.google.cloud.bigquery.TimePartitioning.Type.HOUR
	at java.base/java.lang.Enum.valueOf(Enum.java:240)
	at com.google.cloud.bigquery.TimePartitioning$Type.valueOf(TimePartitioning.java:43)
	at com.google.cloud.bigquery.TimePartitioning.fromPb(TimePartitioning.java:145)
	at com.google.cloud.bigquery.StandardTableDefinition.fromPb(StandardTableDefinition.java:276)
	... 39 more
```

findepi commented:
@losipiuk @ebyhr i put it wrong.
We _do have_ tests. We just do _not_ have testing.

@ebyhr would you be able to add some notes in TestBigQueryIntegrationSmokeTest or in some README
instructing how to run it?

ebyhr commented:
@findepi Sure, basically I just run BigQueryRunner with VM options `-Dbigquery.credentials-file=<path-to-credential-json>`.

wendigo commented:
Ok, so I've tested and there was indeed problem with missing classed/dependencies. I've updated further to version 8.0.0 of BOM and it seems to work fine. I've added that and queried HOUR-ly partitioned table created like this (taken from docs):

```
CREATE TABLE
   test.wendigo (transaction_id INT64,
     transaction_ts TIMESTAMP)
 PARTITION BY
   TIMESTAMP_TRUNC(transaction_ts, HOUR)
 OPTIONS
   ( partition_expiration_days=3,
     description="a table partitioned by transaction_ts" )
```

It's displayed in UI as HOUR-ly partitioned:

![Screenshot 2020-08-26 at 11 45 02](https://user-images.githubusercontent.com/66972/91288813-b7e90f80-e791-11ea-96ca-c91e6586a22c.png)

I've been able to successfully query it using `presto-cli` (first query is before bumping dependencies):

```
presto:test> select * from wendigo;

Query 20200826_093322_00000_kss8b, FAILED, 1 node
Splits: 17 total, 0 done (0.00%)
4.94 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20200826_093322_00000_kss8b failed: com/google/cloud/bigquery/storage/v1beta1/BigQueryStorageGrpc

presto:test> select * from wendigo;
 transaction_id |       transaction_ts
----------------+-----------------------------
           1337 | 2020-08-26 09:25:34.016 UTC
(1 row)

Query 20200826_094100_00000_nm56m, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
6.23 [1 rows, 12B] [0 rows/s, 2B/s]

```

findepi commented:
> I've been able to query HOUR-ly partitioned table created like this (taken from docs):
> 
> ```
> CREATE TABLE
>    test.wendigo (transaction_id INT64,
>      transaction_ts TIMESTAMP)
>  PARTITION BY
>    TIMESTAMP_TRUNC(transaction_ts, HOUR)
>  OPTIONS
>    ( partition_expiration_days=3,
>      description="a table partitioned by transaction_ts" )
> ```
> 
> It's displayed in UI as HOUR-ly partitioned:

Can this be added to `TestBigQueryIntegrationSmokeTest`?

losipiuk commented:
#2390
